shared/constants/transaction.js

```diff
@@ -0,0 +1,95 @@
+// 5. `'submitted'`  - The transaction has been submitted
+// 6. `'failed'`     - The transaction has failed for some reason
+// 7. `'dropped'`    - The transaction was dropped due to a tx with same nonce being accepted
+export const TRANSACTION_STATUS_APPROVED = 'approved'
+export const TRANSACTION_STATUS_CONFIRMED = 'confirmed'
+export const TRANSACTION_STATUS_UNAPPROVED = 'unapproved'
+export const TRANSACTION_STATUS_REJECTED = 'rejected'
+export const TRANSACTION_STATUS_SIGNED = 'signed'
+export const TRANSACTION_STATUS_SUBMITTED = 'submitted'
+export const TRANSACTION_STATUS_FAILED = 'failed'
+export const TRANSACTION_STATUS_DROPPED = 'dropped'
+
+// Transaction Category is a MetaMask construct used internally
+// Used to categorize transactions by purpose for displaying information to the user
+// 1. `'tranfser'`            - A token transaction where the user is sending tokens that they
+//                              own to another address
+// 2. `'transferfrom'`        - A token transaction transferring tokens from an account that
+//                              the sender has an allowance of (see the approve category).
+// 3. `'approve'`             - A token transaction requesting an allowance of the token to
+//                              spend on behalf of the user
+// 4. `'incoming'`            - An incoming (deposit) transaction
+// 5. `'sentEther'`           - A transaction sending ether to a recipient
+// 6. `'contractInteraction'` - A transaction that is interacting with a smart contract's methods
+//                              that we have not treated as a special case, such as approve, transfer
+//                              and transferfrom
+// 7. `'contractDeployment'`  - A transaction that deployed a smart contract
+// 8. `'swap'`                - A transaction swapping one token for another through MetaMask Swaps
+// 9. `'swapApproval'`        - Similar to the approve category, a swap approval is a special case of
+//                              ERC20 approve method that requests an allowance of hte token to spend
+//                              on behalf of the user for the MetaMask Swaps contract. The first swap
+//                              for any token will have an accompanying swapApproval transaction.
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_TRANSFER = 'transfer'
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_APPROVE = 'approve'
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_TRANSFER_FROM = 'transferfrom'
+export const TRANSACTION_CATEGORY_INCOMING = 'incoming'
+export const TRANSACTION_CATEGORY_SENT_ETHER = 'sentEther'
+export const TRANSACTION_CATEGORY_CONTRACT_INTERACTION = 'contractInteraction'
+export const TRANSACTION_CATEGORY_DEPLOY_CONTRACT = 'contractDeployment'
+export const TRANSACTION_CATEGORY_SWAP = 'swap'
+export const TRANSACTION_CATEGORY_SWAP_APPROVAL = 'swapApproval'
+
+// Transaction Group Status is a MetaMask construct to track status of groups
```

Comment:
I am the nit king, I can do anything:
```suggestion
// Transaction Group Status is a MetaMask construct to track the status of groups
```